crates/sui-graphql-rpc/src/types/query.rs

```diff
@@ -48,13 +49,16 @@ pub(crate) type SuiGraphQLSchema = async_graphql::Schema<Query, Mutation, EmptyS
 
 #[Object]
 impl Query {
-    /// First four bytes of the network's genesis checkpoint digest (uniquely identifies the
-    /// network).
     async fn chain_identifier(&self, ctx: &Context<'_>) -> Result<String> {
-        Ok(ChainIdentifier::query(ctx.data_unchecked())
-            .await
-            .extend()?
-            .to_string())
+        let chain_id: ChainIdentifier = *ctx.data()?;
+        // If the chain identifier is not initialized, return an error.
+        if chain_id.0 == NativeChainIdentifier::default() {
```

Comment:
Better to use `Option<NativeChainIdentifier>` here, because technically speaking the all-zero identifier is a valid chain identifier.